### PR TITLE
Invoke to propagate error to caller

### PR DESCRIPTION
**Pull request by anuptalwalkar**
If called function returns an error, we are trying to insert it into the graph (which fails because error is not a pointer type). We should rather just return the error back to the caller.

### CODE REVIEW

File: container.go [hunk 1]
```diff
@@ -67,6 +67,9 @@ func (c *Container) Invoke(t interface{}) error {
 
 		if len(values) > 0 {
 			err, _ := values[len(values)-1].Interface().(error)
```
Comment (alsamylkin) [inline container.go]: coalesce in one line if?

File: container_test.go [hunk 2]
```diff
@@ -384,6 +385,15 @@ func TestInvokeAndRegisterFailure(t *testing.T) {
 	require.Contains(t, err.Error(), "constructor return type must be a pointer")
 }
 
+func TestInvokeReturnedError(t *testing.T) {
+	t.Parallel()
```
Comment (alsamylkin) [inline container_test.go]: one more test, when error is nil?

Comment (anuptalwalkar) [inline container_test.go]: Added

File: container.go [hunk 3]
```diff
@@ -66,16 +66,17 @@ func (c *Container) Invoke(t interface{}) error {
 		values := cv.Call(args)
 
 		if len(values) > 0 {
-			err, _ := values[len(values)-1].Interface().(error)
+			if err, _ := values[len(values)-1].Interface().(error); err != nil {
+				return errors.Wrapf(err, "Error executing the function %v", ctype)
+			}
 			for _, v := range values {
```
Comment (alsamylkin) [inline container.go]: we shouldn't register an error here: just
```golang
for _, v := range values[:len(values]
```
